File: fix_commas.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');

// Function to fix missing commas in object literals and arrays
function fixMissingCommas(content) {
  // Fix object properties missing commas
  content = content.replace(/(\w+):\s*([^,\n}]+)\n\s*(\w+):/g, '$1: $2,\n    $3:');
  content = content.replace(/(\w+):\s*([^,\n}]+)\n\s*}/g, '$1: $2\n  }');
  
  // Fix array elements missing commas
  content = content.replace(/([^,\n\]]+)\n\s*([^,\n\]]+)\n\s*]/g, '$1,\n    $2\n  ]');
  
  // Fix specific patterns
  content = content.replace(/icon: Brain\n\s*title:/g, 'icon: Brain,\n      title:');
  content = content.replace(/title: ([^,\n]+)\n\s*description:/g, 'title: $1,\n      description:');
  content = content.replace(/description: ([^,\n]+)\n\s*benefits:/g, 'description: $1,\n      benefits:');
  
  // Fix metadata objects
  content = content.replace(/title: ([^,\n]+)\n\s*description:/g, 'title: $1,\n  description:');
  content = content.replace(/description: ([^,\n]+)\n\s*keywords:/g, 'description: $1,\n  keywords:');
  content = content.replace(/keywords: ([^,\n]+)\n\s*openGraph:/g, 'keywords: $1,\n  openGraph:');
  
  return content;
}

// Function to process a file
function processFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    const originalContent = content;
    
    // Apply fixes
    content = fixMissingCommas(content);
    
    // Only write if content changed
    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`Fixed: ${filePath}`);
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Function to find all relevant files
function findFiles(dir, extensions = ['.tsx', '.ts', '.js']) {
  const files = [];
  
  function traverse(currentDir) {
    try {
      const items = fs.readdirSync(currentDir);
      
      for (const item of items) {
        const fullPath = path.join(currentDir, item);
        const stat = fs.statSync(fullPath);
        
        if (stat.isDirectory()) {
          // Skip node_modules and .next
          if (!['node_modules', '.next', '.git'].includes(item)) {
            traverse(fullPath);
          }
        } else if (stat.isFile()) {
          const ext = path.extname(item);
          if (extensions.includes(ext)) {
            files.push(fullPath);
          }
        }
      }
    } catch (error) {
      // Skip directories we can't read
    }
  }
  
  traverse(dir);
  return files;
}

// Main execution
function main() {
  const workspaceDir = process.cwd();
  console.log('Starting comma fixing...');
  
  // Find all relevant files
  const files = findFiles(workspaceDir);
  console.log(`Found ${files.length} files to process`);
  
  let fixedCount = 0;
  
  // Process each file
  for (const file of files) {
    if (processFile(file)) {
      fixedCount++;
    }
  }
  
  console.log(`\nFixed ${fixedCount} files`);
  console.log('Comma fixing completed!');
}

// Run the script
if (require.main === module) {
  main();
}

module.exports = { fixMissingCommas, processFile };

File: fix_commas.py
#!/usr/bin/env python3
"""
Fix missing commas in object literals and arrays
"""

import os
import re
import glob

def fix_missing_commas(content):
    """Fix missing commas in object literals and arrays"""
    lines = content.split('\n')
    fixed_lines = []
    
    for i, line in enumerate(lines):
        # Check if this line ends an object property or array element
        if re.match(r'^\s*[a-zA-Z_][a-zA-Z0-9_]*\s*:', line) or re.match(r'^\s*"[^"]*"\s*$', line) or re.match(r'^\s*[0-9]+\s*$', line):
            # Check if next line starts a new property/element
            if i + 1 < len(lines):
                next_line = lines[i + 1]
                if (re.match(r'^\s*[a-zA-Z_][a-zA-Z0-9_]*\s*:', next_line) or 
                    re.match(r'^\s*"[^"]*"\s*$', next_line) or 
                    re.match(r'^\s*[0-9]+\s*$', next_line) or
                    re.match(r'^\s*}\s*$', next_line) or
                    re.match(r'^\s*]\s*$', next_line)):
                    # Add comma if missing
                    if not line.rstrip().endswith(',') and not line.rstrip().endswith('{') and not line.rstrip().endswith('['):
                        line = line.rstrip() + ','
        
        fixed_lines.append(line)
    
    return '\n'.join(fixed_lines)

def fix_file(file_path):
    """Fix a single file"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        content = fix_missing_commas(content)
        
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed: {file_path}")
            return True
        else:
            print(f"No changes needed: {file_path}")
            return False
            
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    """Main function to fix all files"""
    files_to_fix = [
        'app/page.tsx',
        'app/team/page.tsx',
        'app/layout.tsx'
    ]
    
    fixed_count = 0
    for file_path in files_to_fix:
        if os.path.isfile(file_path):
            if fix_file(file_path):
                fixed_count += 1
    
    print(f"\nFixed {fixed_count} files out of {len(files_to_fix)} checked")

if __name__ == "__main__":
    main()